adr/ADR-48.md

```diff
@@ -0,0 +1,71 @@
+# JetStream Distributed Counter CRDT
+
+| Metadata | Value      |
+|----------|------------|
+| Date     | 2025-01-09 |
+| Author   | @ripienaar |
+| Status   | Approved   |
+| Tags     | jetstream  |
+
+| Revision | Date       | Author     | Info                    |
+|----------|------------|------------|-------------------------|
+| 1        | 2025-01-09 | @ripienaar | Document Initial Design |
+
+# Context and Motivation
+
+We wish to provide a distributed counter that will function in Clusters, Super Clusters, through Mirrors and Sources
+and any other way that data might reach a Stream.
+
+We will start with basic addition and subtraction primitives which will automatically behave as a CRDT and be order independent.
+
+# Solution Overview
+
+A Stream can opt-in to supporting Counters which will allow any key to be a counter.
+
+Publishing a message to such a Stream will load the most recent message on the subject, increment it's value and save the new message with the latest value.
+
+```bash
+$ nats s get COUNTER --last-for counter.hits
+Item: COUNTER#22802062 received 2025-01-09 18:05:07.93747413 +0000 UTC on Subject counter.hits
+
+Headers:
+  Nats-Incr: +2
+
+{"val":100}
+$ nats pub counter.hits '' -J -H "Nats-Incr:+1"
+$ nats s get COUNTER --last-for counter.hits
+Item: COUNTER#22802063 received 2025-01-09 18:06:00 +0000 UTC on Subject counter.hits
+
+Headers:
+  Nats-Incr: +1
+
+{"val":101}
```

Comment:
Suggestions on what else to put in this JSON here - though less is more.  Will then after discussion document it here also

Reply:
Should the value be a float64 so that one can add `1.23` to the value stored?

Reply:
I like starting simple.